src/frontend/screens/LeaveProject/LeaveProject.tsx

```diff
@@ -0,0 +1,200 @@
+import HeaderTitle from "../../sharedComponents/HeaderTitle";
+import ObservationHeaderRight from "../Observation/ObservationHeaderRight";
+import AppStack from "../../AppStack";
+
+const m = defineMessage({
+  leaveProjectTitle: {
+    id: "screens.LeaveProject.LeaveProject.leaveProjectTitle",
+    defaultMessage: "Leave Project{projectName}?",
+  },
+  headerTitle: {
+    id: "screens.LeaveProject.LeaveProject.headerTitle",
+    defaultMessage: "Leave Project",
+  },
+  willDelete: {
+    id: "screens.LeaveProject.LeaveProject.willDelete",
+    defaultMessage:
+      "This will delete {numOfObservations} observations and {numOfPics} Pictures",
+  },
+  agreeToDelete: {
+    id: "screens.LeaveProject.LeaveProject.agreeToDelete",
+    defaultMessage: "I understand I will be deleting all data from my device.",
+  },
+  leaveButton: {
+    id: "screens.LeaveProject.LeaveProject.leaveButton",
+    defaultMessage: "Leave Project",
+  },
+  cancel: {
+    id: "screens.LeaveProject.LeaveProject.cancel",
+    defaultMessage: "Cancel",
+  },
+  confirmDelete: {
+    id: "screens.LeaveProject.LeaveProject.confirmDelete",
+    defaultMessage: "Please check the box to confirm",
+  },
+  syncWarning: {
+    id: "screens.LeaveProject.LeaveProject.syncWarning",
+    defaultMessage:
+      "Sync with a team member so you don't loose all your observations.",
+  },
+});
+
+const navOptions: NavigationStackOptions = {
+  headerTitle: () => (
+    <HeaderTitle>
+      <FormattedMessage {...m.headerTitle} />
+    </HeaderTitle>
+  ),
+};
+
+export const LeaveProjectScreen: NavigationStackScreenComponent = () => {
+  const [confirmDelete, setConfirmDelete] = React.useState(false);
+  const [pristine, setPristine] = useState(true);
+  const [config] = useContext(ConfigContext);
+
+  const nav = useNavigation();
+
+  function getObservationData() {}
+
+  function leaveProject() {
+    if (pristine) setPristine(false);
+
+    if (!confirmDelete) return;
+
+    nav.navigate("LeaveProjectProgress");
+  }
+
+  const name = config.metadata.name ? " " + config.metadata.name : "";
+
+  return (
+    <View style={styles.screenContainer}>
+      <View>
+        <Text style={styles.title}>
+          <FormattedMessage
+            {...m.leaveProjectTitle}
+            values={{ projectName: name }}
+          />
+        </Text>
+
+        <Text style={styles.subHeader}>
+          <FormattedMessage {...m.willDelete} />
+        </Text>
+      </View>
+
+      <View style={styles.warningContainer}>
+        <Image
+          source={require("../../images/Vector.png")}
+          style={{ margin: 10 }}
+        />
+        <Text style={{ padding: 5, flex: 1 }}>
+          <FormattedMessage {...m.syncWarning} />
+        </Text>
+      </View>
+
+      <View style={styles.confirmContainer}>
+        <View style={styles.checkContainer}>
+          <CheckBox
+            value={confirmDelete}
+            onValueChange={() => setConfirmDelete(!confirmDelete)}
```

Comment:
nitpick but i generally lean towards the [functional update](https://reactjs.org/docs/hooks-reference.html#functional-updates) syntax when it comes to toggling the value i.e.

```js
setConfirmDelete(previous => !previous)
```